.github/workflows/build.yaml

```diff
@@ -4,27 +4,28 @@ on:
   push:
     branches: [ rolling ]
   workflow_dispatch:
+  schedule:
+    # Run every morning to detect flakiness and broken dependencies
+    - cron: '03 5 * * *'
 defaults:
   run:
     shell: bash
 jobs:
-  test:
-    runs-on: ubuntu-latest
-    strategy:
-      fail-fast: false
-      matrix:
-        distro: ['iron', 'jazzy', 'rolling']
-    container:
-      image: ros:${{ matrix.distro }}-ros-base
-    timeout-minutes: 30
-    steps:
-    - name: Deps
-      run: |
-        apt update && apt install -y curl
-    - uses: actions/checkout@v4
-    - name: rosdep
-      run: |
-        rosdep update
-        rosdep install --from-paths . -yir
-    - name: build
-      run: /ros_entrypoint.sh colcon build
+  build_and_test:
+      name: build_and_test_binaries_${{ matrix.ROS_DISTRO }}
+      strategy:
+        matrix:
+          ROS_DISTRO: [rolling, jazzy, iron]
+      runs-on: ubuntu-latest
+      container:
+        image: rostooling/setup-ros-docker:ubuntu-noble-latest
```

Comment:
We can't alwayse use `noble` since iron is distributed on `jammy`. Please see my previous comment on how the matrix can be updated to include distro and os. https://github.com/ros2/rmw_zenoh/pull/256#discussion_r1727744706

Not addressed form the same comment:
* Using If condition to include `vcs-repo-file-url` 
* Using ros-tooling/setup-ros@v0.7 instead of the `rostooling/setup-ros-docker:ubuntu` image

Reply:
Firstly sorry for some commits, but would using separate jobs for source and bineries be good?  I am yet to complete my work. I will ping you.

Reply:
I'd prefer to have everything in the same job. The CI here is a temporary measure until we start running the usual CI jobs using the ROS buildfarm.

Reply:
I've applied @Yadunund and i've readded the fail-fast param which @ahcorde has overlooked by deleting this. Would you review in your available time?

Reply:
thank you for the changes @CihatAltiparmak